python/pyarrow/table.pxi

```diff
@@ -625,6 +629,12 @@ cdef class RecordBatch(_PandasConvertible):
             size += self.column(i).nbytes
         return size
 
+    def __sizeof__(self):
+        return (
+            object.__sizeof__(self) + self.nbytes +
+            sys.getsizeof(self.schema.metadata)
```

Comment:
I don't think it makes sense to add the schema metadata size, since we are not adding the schema size.

Reply:
Shouldn't we add the schema size? At least in some cases the schema can become quite big.

Reply:
It's a bit of a fine line, but generally `__sizeof__` shouldn't include the size of objects which are reachable individually. For example:
```python
>>> a = np.zeros(0)                                                                       
>>> sys.getsizeof(a)                                                                      
96
>>> sys.getsizeof(a.dtype)                                                                
96
```

Reply:
Ah, I had the misconception that `sys.getsizeof` would be recursive. Then we also shouldn't include the metadata size here, too.

Reply:
> It's a bit of a fine line, but generally __sizeof__ shouldn't include the size of objects which are reachable individually

You could say they same about the `buffers` then ... as those are also reachable individually (just to illustrate the thin line ;) I don't want to argue for not including those). 

It's true you can access the schema / schema metadata through a python API, but whether you access this or not, the actual schema (metadata) is stored anyway in the Table (C++) object, so I don't find it necessarily clear on which side of the line it falls.

Reply:
Yes, but the `buffers` are not really useful in isolation, while the schema can definitely be applied to multiple tables or batches (and often will). So I'd argue that not including the schema is the most useful here.

Reply:
OK! What about implementing a `__sizeof__` for schema that includes the metadata? Or can metadata be shared across different schema's as well?

Reply:
> What about implementing a __sizeof__ for schema that includes the metadata?

That sounds fine to me. The metadata can be shared at the C++ level, but I'm not sure that's common.